app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt

```diff
@@ -206,118 +207,289 @@ class OptionsFragmentTest {
+          hasExtra(
+            ReadingTextSizeActivity.KEY_READING_TEXT_SIZE_PREFERENCE_SUMMARY_VALUE,
+            "Medium"
+          ),
+          hasComponent(ReadingTextSizeActivity::class.java.name)
+        )
       )
+    }
+  }
 
-      it.onActivity { activity ->
-        activity.startActivityForResult(
-          createAppLanguageActivityIntent(summaryValue = "English"),
-          REQUEST_CODE_APP_LANGUAGE
+  @Test
+  fun openOptionsActivity_selectAppLanguage_showAppLanguageActivitySuccessfully() {
+    launch<OptionsActivity>(
+      createOptionActivityIntent(
+        internalProfileId = 0,
+        isFromNavigationDrawer = true
+      )
+    ).use {
+      onView(
+        atPositionOnView(
+          recyclerViewId = R.id.options_recyclerview,
+          position = 1,
+          targetViewId = R.id.app_language_text_view
+        )
+      ).perform(click())
+      intended(
+        allOf(
+          hasExtra(
+            AppLanguageActivity.APP_LANGUAGE_PREFERENCE_SUMMARY_VALUE_EXTRA_KEY,
+            "English"
+          ),
+          hasComponent(AppLanguageActivity::class.java.name)
         )
-      }
+      )
+    }
+  }
 
-      getInstrumentation().waitForMonitorWithTimeout(activityMonitor, 3)
-      testCoroutineDispatchers.runCurrent()
+  @Test
+  fun openOptionsActivity_configChange_selectAppLanguage_showAppLanguageActivitySuccessfully() {
+    launch<OptionsActivity>(
+      createOptionActivityIntent(
+        internalProfileId = 0,
+        isFromNavigationDrawer = true
+      )
+    ).use {
+      onView(isRoot()).perform(orientationLandscape())
       onView(
         atPositionOnView(
           recyclerViewId = R.id.options_recyclerview,
           position = 1,
           targetViewId = R.id.app_language_text_view
         )
-      ).check(
-        matches(withText("French"))
+      ).perform(click())
+      intended(
+        allOf(
+          hasExtra(
+            AppLanguageActivity.APP_LANGUAGE_PREFERENCE_SUMMARY_VALUE_EXTRA_KEY,
+            "English"
+          ),
+          hasComponent(AppLanguageActivity::class.java.name)
+        )
+      )
+    }
+  }
+
+  @Test
+  fun openOptionsActivity_selectAudioLanguage_showAudioLanguageActivitySuccessfully() {
+    launch<OptionsActivity>(
+      createOptionActivityIntent(
+        internalProfileId = 0,
+        isFromNavigationDrawer = true
+      )
+    ).use {
+      onView(
+        atPositionOnView(
+          recyclerViewId = R.id.options_recyclerview,
+          position = 2,
+          targetViewId = R.id.audio_language_text_view
+        )
+      ).perform(click())
+      intended(
+        allOf(
+          hasExtra(
+            AudioLanguageActivity.KEY_AUDIO_LANGUAGE_PREFERENCE_SUMMARY_VALUE,
+            "English"
+          ),
+          hasComponent(AudioLanguageActivity::class.java.name)
+        )
+      )
+    }
+  }
+
+  @Test
+  fun openOptionsActivity_configChange_selectAudioLanguage_showAudioLanguageActivitySuccessfully() {
```

Comment:
Rename to `openOptionsActivity_configChange_clickAudioLanguage_opensAudioLanguageActivitySuccessfully`

Reply:
Done.